mmengine/hooks/checkpoint_hook.py

```diff
@@ -304,6 +327,47 @@ def after_val_epoch(self, runner, metrics):
 
         self._save_best_checkpoint(runner, metrics)
 
+    def after_train(self, runner) -> None:
+        """Publish the checkpoint after training.
+
+        Args:
+            runner (Runner): The runner of the training process.
+        """
+        if self.published_keys is None:
+            return
+
+        if self.save_last and 'last_ckpt' in self.runtime_info:
+            last_ckpt = runner.message_hub.get_info('last_ckpt')
+            self._publish_model(runner, last_ckpt)
+
+        if self.save_best is not None:
+            if len(self.key_indicators) == 1:
+                if self.best_ckpt_path is not None:
+                    self._publish_model(runner, self.best_ckpt_path)
+            else:
+                for key, best_ckpt in self.best_ckpt_path_dict.items():
+                    self._publish_model(runner, best_ckpt)
```

Comment:
Will this format more clear?
```suggestion
        if getattr(self, 'best_ckpt_path', None) is not None:
            self._publish_model(runner, self.best_ckpt_path)
        if getattr(self, 'best_ckpt_path_dict', None) is not None:
            for key, best_ckpt in self.best_ckpt_path_dict.items():
                self._publish_model(runner, best_ckpt)
```